src/main/java/org/broadinstitute/hellbender/tools/spark/validation/BqsrRecalBasesCompare.java

```diff
@@ -0,0 +1,230 @@
+import org.apache.spark.api.java.function.Function2;
+import org.apache.spark.api.java.function.PairFunction;
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.cmdline.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.programgroups.SparkProgramGroup;
+import org.broadinstitute.hellbender.engine.spark.GATKSparkTool;
+import org.broadinstitute.hellbender.engine.spark.datasources.ReadsSparkSource;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.QualityUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import scala.Tuple2;
+
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+@CommandLineProgramProperties(summary = "Compares two the quality scores of BAMs", oneLineSummary = "Diff qs of two BAMs",
+        programGroup = SparkProgramGroup.class)
+
+public class BqsrRecalBasesCompare extends GATKSparkTool  {
+    private static final long serialVersionUID = 1L;
+
+    @Override
+    public boolean requiresReads() { return true; }
+
+    @Argument(doc="the second BAM", shortName = "I2", fullName = "input2", optional = false)
+    protected String input2;
+
+    @Argument(doc="print matrix", shortName = "pm", fullName = "printMatrix", optional = true)
+    protected Boolean printMatrix = false;
+
+    @Argument(doc="print summary", shortName = "ps", fullName = "printSummary", optional = true)
+    protected Boolean printSummary = true;
+
+    @Argument(doc="crash on diff", shortName = "cd", fullName = "crashOnDiff", optional = true)
+    protected Boolean crashOnDiff = false;
+
+    @Override
+    protected void runTool(final JavaSparkContext ctx) {
+        JavaRDD<GATKRead> firstReads = getReads();
+        ReadsSparkSource readsSource2 = new ReadsSparkSource(ctx);
+        JavaRDD<GATKRead> secondReads = readsSource2.getParallelReads(input2, getIntervals());
+
+        long firstBamSize = firstReads.count();
+        long secondBamSize = secondReads.count();
+        if (firstBamSize != secondBamSize) {
+            throw new UserException("input bams have different numbers of total reads: "
+                    + firstBamSize + "," + secondBamSize);
+        }
+        System.out.println("Processing bams with " + firstBamSize + " reads");
+
+        // To compare bases, we created two PairRDDs keyed by read name, strand, etc. where the value is the bases.
+        // We cogroup the two RDDs, verify there is only one read per key per bam, and then diff the bases
+        // position by position for the two reads and put the results in CompareMatrix.
+        JavaPairRDD<String, Quals> firstQuals = firstReads.mapToPair(read -> {
+            read.clearAttributes();
+            return new Tuple2<>(read.toString(), new Quals(read.getBaseQualities()));
```

Comment:
`toString()` just for display (only contains name and position) -- use `getSAMString()` instead